Components/Terrain/src/OgreTerrainGroup.cpp

```diff
@@ -309,11 +308,11 @@ namespace Ogre
             LoadRequest req;
             req.slot = slot;
             req.origin = this;
-            ++LoadRequest::loadingTaskNum;
-            Root::getSingleton().getWorkQueue()->addRequest(
-                mWorkQueueChannel, WORKQUEUE_LOAD_REQUEST, 
-                Any(req), 0, synchronous);
-
+            WorkQueue::RequestID id =
+                Root::getSingleton().getWorkQueue()->addRequest(
+                    mWorkQueueChannel, WORKQUEUE_LOAD_REQUEST,
+                    Any(req), 0, synchronous);
+            mTerrainPrepareRequests.insert(TerrainPrepareRequestMap::value_type(slot, id));
```

Comment:
This is the problem line. When synchronous is true it won't go into background thread and there won't be anything in `mTerrainPrepareRequests` so it will be deleted.

Reply:
local fix:
```cpp
     TerrainPrepareRequestMap::iterator iter =
                mTerrainPrepareRequests.insert(TerrainPrepareRequestMap::value_type(slot, 0)).first;
            iter->second = Root::getSingleton().getWorkQueue()->addRequest(
                mWorkQueueChannel, WORKQUEUE_LOAD_REQUEST, Any(req), 0, synchronous);
```

Reply:
You can't insert to that iterator at that point anymore since it will be invalid at that point. This is how I'd fix it:
```
            std::pair<TerrainPrepareRequestMap::iterator, bool> ret = mTerrainPrepareRequests.insert(TerrainPrepareRequestMap::value_type(slot, 0));
            assert(ret.second == false);
            WorkQueue::RequestID id =
                Root::getSingleton().getWorkQueue()->addRequest(
                    mWorkQueueChannel, WORKQUEUE_LOAD_REQUEST,
                    Any(req), 0, synchronous);
            if (!synchronous)
                ret.first->second = id;
```

Reply:
Ok this fix stops `Terrain` sample from crashing at start.

Reply:
ah.. yes.. I figured. seems to be properly fixed this way